app/src/lib/error-with-metadata.ts

```diff
@@ -1,18 +1,32 @@
+import { Repository } from '../models/repository'
+import { RetryAction } from './retry-actions'
+import { CloningRepository } from './dispatcher/cloning-repositories-store'
+
 export interface IErrorMetadata {
   /** Was the action which caused this error part of a background task? */
   readonly backgroundTask?: boolean
+
+  /** The repository from which this error originated. */
+  readonly repository?: Repository | CloningRepository
+
+  /** The action to retry, if possible. */
```

Comment:
I think this is saying that the retry action should only be specified if retrying is possible.